verification/src/http_server/handlers/verification/sourcify/mod.rs

```diff
@@ -1,53 +1,20 @@
+mod api;
+mod metadata;
 mod types;
 
-use self::types::{ApiRequest, ApiVerificationResponse};
-use crate::{Config, VerificationResult};
+use self::api::{verify_using_sourcify_client, SoucifyApiClient};
+use self::types::ApiRequest;
+use crate::Config;
 use actix_web::web;
-use actix_web::{error, error::Error, web::Json};
-use std::collections::HashMap;
+use actix_web::{error::Error, web::Json};
 
 use super::VerificationResponse;
 
-async fn verification_request(
-    config: &Config,
-    params: &ApiRequest,
-) -> Result<Json<VerificationResponse>, Error> {
-    let resp = reqwest::Client::new()
-        .post(&config.sourcify.api_url)
-        .json(&params)
-        .send()
-        .await
-        .map_err(error::ErrorInternalServerError)?;
-
-    let response_body: ApiVerificationResponse =
-        resp.json().await.map_err(error::ErrorInternalServerError)?;
-
-    match response_body {
-        ApiVerificationResponse::Verified { result } => {
-            // TODO: parse metadata.json, return abi, constructor arguments, ...
-            let _ = result;
-            let verification_result = VerificationResult {
-                contract_name: "".into(),
-                compiler_version: "".into(),
-                evm_version: "".into(),
-                constructor_arguments: None,
-                contract_libraries: None,
-                abi: "".into(),
-                sources: HashMap::new(),
-            };
-            Ok(Json(VerificationResponse::ok(verification_result)))
-        }
-        ApiVerificationResponse::Error { error } => Ok(Json(VerificationResponse::err(error))),
-        ApiVerificationResponse::ValidationErrors { message, errors } => {
-            let error_message = format!("{}: {:?}", message, errors);
-            Ok(Json(VerificationResponse::err(error_message)))
-        }
-    }
-}
-
 pub async fn verify(
     config: web::Data<Config>,
     params: Json<ApiRequest>,
 ) -> Result<Json<VerificationResponse>, Error> {
-    verification_request(config.as_ref(), &params.into_inner()).await
+    let sourcify_client = SoucifyApiClient::new(&config.sourcify.api_url);
```

Comment:
I suggest to initialize `SourciftApiClient` only once (e.g. while configuring the corresponding route) and pass it as a parameter instead of config